rbd: return crypt error for the rpc return

humblec opened the pull request:
at present we return the volume connect error if the clone
from snapshot fails when rbdvolume is encrypted, which is incorrect.
This patch correctly return the failed copy encryption error to the
caller

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

nixpanic reviewed (changes requested):
Instead of using `cryptErr`, it probably is just as simple to use `err` everywhere. There is no other contents in `err` anyway. Also, consider removing the log.

I think this was copy/pasted from an other location where there could have been two different errors.

humblec commented:
> Instead of using `cryptErr`, it probably is just as simple to use `err` everywhere. There is no other contents in `err` anyway. Also, consider removing the log.
> Instead of using `cryptErr`, it probably is just as simple to use `err` everywhere. There is no other contents in `err` anyway. Also, consider removing the log.
> 

> 
> I think this was copy/pasted from an other location where there could have been two different errors.

@nixpanic the cloneFromSnapshot() make use of a new error var uErr for undoSnapshotCloning() just above this one and doSnapshot() also distinguish the error to a new var like cryptErr in the same source file. so , imo, keeping a clear or in parity error variable would be better. wdyt? 

I agree we can remove the extra logging :+1: 

> I think this was copy/pasted from an other location where there could have been two different errors.

nixpanic commented:
> @nixpanic the cloneFromSnapshot() make use of a new error var uErr for undoSnapshotCloning() just above this one and doSnapshot() also distinguish the error to a new var like cryptErr in the same source file. so , imo, keeping a clear or in parity error variable would be better. wdyt?

I prefer to keep things simple. Having a single error variable like we do in most cases is the easiest to understand, and that makes it more difficult to make mistakes like the one fixed in this PR. In the cases where two errors are possible, the 2nd error should be logged, and the 1st error returned. That obviously requires two errors, but all other cases should be able to use a single one.

humblec committed:
rbd: return crypt error for the rpc return

At present we return the volume connect error if the clone
from snapshot fails when rbdvolume is encrypted, which is incorrect.
This patch correctly return the failed copy encryption error to the
caller

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec commented:
@nixpanic comments addressed . PTAL. Thanks.

Madhu-1 commented on internal/rbd/controllerserver.go:
```diff
@@ -811,13 +811,9 @@ func cloneFromSnapshot(ctx context.Context, rbdVol *rbdVolume, rbdSnap *rbdSnaps
-				err.Error())
+		err = rbdVol.copyEncryptionConfig(&vol.rbdImage)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, err.Error())
```
as no formatting is required, it should be status.Error()

humblec replied:
@Madhu-1 sure.. done.. ptal. thanks

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.21

nixpanic commented:
> /test ci/centos/mini-e2e/k8s-1.21

Just running for stability check, see #1963
